### PR TITLE
Add ability to assign case

### PR DESCRIPTION
**Pull request by willmcb**
## Description
Adds ability to assign case. 

Will raise another PR to deal with assigning cases as trigger.

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### CODE REVIEW

**Review by digitalali-moj — COMMENTED** [review]
f/e looks good to me, thanks :)

File: app/views/cases/sar_internal_review/_case_details.html.slim [hunk 1]
```diff
@@ -1 +1,32 @@
...
+        = render partial: 'cases/shared/case_deadlines', locals: { case_details: case_details }
+        = render partial: 'cases/sar/response_address', locals: { case_details: case_details }
+        - if case_details.has_date_draft_compliant?
+          = render partial: 'cases/shared/draft_compliance_details', locals: { case_details: case_details }
+        - if case_details.responding_team.present?
+          = render partial: 'cases/shared/responding_team', locals: { case_details: case_details }
+        - if case_details.closed?
+          = render partial: 'cases/shared/closed_case_details', locals: { case_details: case_details }
+
+    - if @case.type_abbreviation != case_details.type_abbreviation
```
Comment (ymao2) [inline app/views/cases/sar_internal_review/_case_details.html.slim:27]: Sorry, this line looks very suspicious to me,  are you trying to share this page between another case type? not sure under which scenario the case_detail will have different case type with @case, especially you place this under SAR_IR which is the leaf class already.  

Also if you just want to show the linked original case, there is already a shared template page for it 
shared/components/case_linked_cases,  you can check it from show.html.slim
but I may misunderstand your intention, let me know, if you need to discuss it

Comment (ymao2) [reply app/views/cases/sar_internal_review/_case_details.html.slim]: @willmcb , tbh, I am not fully sure  whether the line of code will cause trouble or not, but at least the linked cases should show up automatically from link-cases-section.  we can always add it back if somehow we need this one in the future

**Review by ymao2 — APPROVED** [review]
LGTM :)

File: app/views/cases/sar_internal_review/_case_details.html.slim (hunk 1 above)
Comment (willmcb) [reply app/views/cases/sar_internal_review/_case_details.html.slim]: 👍

**Pull request merged**